EqualityAsserts.cs

```diff
@@ -9,6 +9,10 @@
 using System.Collections.Generic;
 using System.Globalization;
 using Xunit.Sdk;
+#if XUNIT_ASSERT_ARRAY_AS_SPAN
```

Comment:
This is not necessary. Based on https://www.nuget.org/packages/system.memory it appears that Span can be made functional everywhere that we support assertions.

Reply:
Actually, I take it back. There are things here that require C# 7.3 so we do still need this. That said, there will be more Span-related things coming, so let's change this to `XUNIT_SPAN`, and please move it up to the top as a separate block, like you see I've done for `XUNIT_NULLABLE`.

Reply:
It turns out I don't need that block after all.